test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -12527,6 +12527,44 @@ BOOST_AUTO_TEST_CASE(abi_encode_empty_string_v2)
 		0x00
 	));
 }
+
+BOOST_AUTO_TEST_CASE(abi_encode_rational)
+{
+	char const* sourceCode = R"(
+		// Tests that rational numbers (even negative ones) are encoded properly.
+		contract C {
+			function f() public pure returns (bytes memory, bytes memory) {
+				return (abi.encode(1, -2), abi.encodePacked(int8(1), int8(-2)));
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "C");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(
+		0x40, 0xa0,
+		0x40, u256(1), u256(-2),
+		0x20, bytes{0x01, 0xfe}
+	));
+}
+
+BOOST_AUTO_TEST_CASE(abi_encode_rational_v2)
+{
+	char const* sourceCode = R"(
+		// Tests that rational numbers (even negative ones) are encoded properly.
+		pragma experimental ABIEncoderV2;
+		contract C {
+			function f() public pure returns (bytes memory, bytes memory) {
+				return (abi.encode(1, -2), abi.encodePacked(int8(1), int8(-2)));
```

Comment:
I guess we could remove the `abi.encodePacked` part because it doesn't test the case.